Tiqr/Classes/AuthenticationSummaryViewController.m

```diff
@@ -44,15 +45,17 @@ @interface AuthenticationSummaryViewController ()
 @property (nonatomic, strong) IBOutlet UILabel *serviceProviderDisplayNameLabel;
 @property (nonatomic, strong) IBOutlet UILabel *serviceProviderIdentifierLabel;
 @property (nonatomic, strong) IBOutlet UIButton *returnButton;
+@property (nonatomic, copy) NSString *PIN;
 
 @end
 
 @implementation AuthenticationSummaryViewController
 
-- (instancetype)initWithAuthenticationChallenge:(AuthenticationChallenge *)challenge {
+- (instancetype)initWithUsedPIN:(NSString *)PIN {
     self = [super initWithNibName:@"AuthenticationSummaryView" bundle:nil];
 	if (self != nil) {
-		self.challenge = challenge;
+		self.challenge = ServiceContainer.sharedInstance.challengeService.currentAuthenticationChallenge;
+        self.PIN = PIN;
```

Comment:
Maybe this is avoidable by always performing the upgrade to TouchID after a successful sign in, while keeping the stored data in the keychain. When the user has decided if he wants to upgrade or not, the obsolete data can be deleted.